### PR TITLE
Fixes #24814 - fix tests and unify JSON fixtures

### PR DESCRIPTION
**Pull request by lzap**
This is just a refactoring of tests, but it adds one extra test which is currently failing until the following PR is merged in core: https://github.com/theforeman/foreman/pull/5989

EDIT: Develop is broken, so I've incorporated fixes for React menu and template locking.

### CODE REVIEW

**Comment by lzap** [comment]
Rebased @rahulbajaj0509 want to do quick look and merge?

File: test/unit/host_discovered_test.rb [hunk 1]
```diff
@@ -380,39 +380,46 @@ class HostDiscoveredTest < ActiveSupport::TestCase
...
+    raw = parse_json_fixture('regular_host', true)
     host = discover_host_from_facts(raw)
     refute_nil host.primary_interface
     assert_equal "eth0", host.primary_interface.identifier
   end
 
+  test "primary interface via tagged VLAN is discovered with an IP" do
+    raw = parse_json_fixture('rhel7-vlan')
+    host = discover_host_from_facts(raw)
+    assert_equal "192.168.1.16", host.ip
```
Comment (rabajaj0509) [inline test/unit/host_discovered_test.rb]: shouldn't the line above have `host.primary_interface.ip`?

Comment (lzap) [reply test/unit/host_discovered_test.rb]: I should have been more verbose - this test will fail until the core PR is merged as stated in the description. Sorry, I will remove this from this PR and move it somewhere else.

**Comment by lzap** [comment]
Removed the failing test from this refactoring, created new PR.

**Comment by lzap** [comment]
[test discovery]

**Comment by lzap** [comment]
Develop is broken, so I've incorporated fixes for React menu and template locking.

**Comment by rabajaj0509** [comment]
works flawlessly, this is too much work! thanks @lzap :)

**Pull request merged**